### DISCUSSION TITLE
Side effect detection

### DISCUSSION THREAD

**Discussion by jmglov** [repl-acement/repl-acement#28] [Ideas]
It just occurred to me yesterday that if REPL-acement had a list of side-effecting functions, it could mark functions as pure or not, and serve as the basis for static analysis tools that could warn you about possible attack vectors. This is obviously heuristical to some degree, as you can always interop with some native code that has side effects, but it might be an idea worth exploring more.

**Comment by raymcdermott** [comment]
Yes! There is a lot work going on around limiting ambient authority, especially to 3rd party code.

Maybe we can use the permission system of Deno as inspiration 

https://deno.land/manual@v1.17.1/getting_started/permissions

**Comment by raymcdermott** [comment]
Also SCI has a whitelist of permitted classes that can be used to limit entry
